code/game/objects/objs.dm

```diff
@@ -135,17 +135,32 @@
 	handle_rotation()
 	return buckled_mob
 
-/obj/proc/unbuckle()
-	if(buckled_mob)
-		if(buckled_mob.buckled == src)	//this is probably unneccesary, but it doesn't hurt
-			buckled_mob.buckled = null
-			buckled_mob.anchored = initial(buckled_mob.anchored)
-			buckled_mob.update_canmove()
 
-			var/M = buckled_mob
-			buckled_mob = null
-			UnregisterSignal(M, COMSIG_LIVING_DO_RESIST)
-			afterbuckle(M)
+/obj/proc/unbuckle(mob/user, silent = TRUE)
+	if(!buckled_mob || buckled_mob.buckled != src)
+		CRASH("[src] called unbuckle() with [user ? user : "no"] user and [buckled_mob ? (buckled_mob.buckled ? "with [buckled_mob.buckled] as buckled_mob.buckled" : "no buckled_mob.buckled") : "no buckled_mob"].")
```

Comment:
Gonna wait a bit before merging this to see if this produces any unwanted runtimes.

Reply:
It did and they got fixed.

Reply:
Still gonna give it a few days, then it can get removed.

Reply:
Fuck it, I can live with a single CASH in my life

Reply:
Remind me to open a PR to remove it.
I was waiting until it had been considered tested enough.